x-pack/legacy/plugins/ml/public/components/job_selector/job_selector.js

```diff
@@ -139,12 +139,21 @@ export function JobSelector({
     handleResize();
   }, [handleResize, jobs]);
 
-  function closeFlyout() {
+  // On opening and closing the flyout, optionally update a global `skipRefresh` flag.
+  // This allows us to circumvent race conditions which could happen by triggering both
+  // timefilter and job selector related events in Single Metric Viewer.
+  function closeFlyout(setSkipRefresh = true) {
     setIsFlyoutVisible(false);
+    if (setSkipRefresh) {
+      setGlobalStateSkipRefresh(globalState, false);
+    }
   }
 
-  function showFlyout() {
+  function showFlyout(setSkipRefresh = true) {
```

Comment:
Wouldn't it be more intuitive to have `setSkipRefresh = false` and then send in `true` when we want to set it? 🤔

Reply:
My intention was to have it default to `true` because it is the default behaviour and therefore in the usual cases where it's used and e.g. passed on as a prop like `onClose={closeFlyout}` doesn't need wrapping in another callback and no code change to set it to true. The only time we need it to be `false` is inside `applySelection` where we call `closeFlyout` directly and can easily pass on the flag.